src/main/java/study/CalcImpl.java

```diff
@@ -0,0 +1,46 @@
+package study;
+
+public class CalcImpl implements Calc {
+
+    @Override
+    public int cal(String input) {
+        preconditions(input);
+        String[] splited = input.split(" ");
+        int initValue = Integer.valueOf(splited[0]);
+        for (int i = 1; i < splited.length; i = i + 2) {
+            initValue = doCal(initValue, Integer.valueOf(splited[i + 1]), splited[i]);
+        }
+        return initValue;
+    }
+
+    private void preconditions(String input) {
+        if(input == null || input.trim().equals("")) throw new IllegalArgumentException();
```

Comment:
```suggestion
        if (Objects.isNull(input) || input.trim().equals("")) {
            throw new IllegalArgumentException();
        }
```
Objects.isNull()을 사용하여 가독성을 높여보세요. 또한 중괄호를 생략하지 않는것이 어떨까요 ?

Reply:
`""` 와 같은 값은 상수로 선언하여 관리해보는것이 어떨까요 ? 
https://www.slipp.net/questions/356